support/k8s/k8s-workload-registrar/README.md

```diff
@@ -0,0 +1,130 @@
+defines the `example-cluster`, the following node registration entry would
+be created and used as the parent for all workloads:
+
+```
+Entry ID      : 7f18a693-9f94-4e91-af7a-a8a61e9f4bce
+SPIFFE ID     : spiffe://example.org/k8s-workload-registrar/example-cluster/node
+Parent ID     : spiffe://example.org/spire/server
+TTL           : default
+Selector      : k8s_psat:cluster:example-cluster
+```
+
+## Workload Registration
+
+The registrar handles pod CREATE and DELETE admission review requests to create
+and delete registration entries for workloads running on those pods. The
+workload registration entries are configured to run on any node in the
+cluster.
+
+There are two workload registration modes.
+
+### Service Account Based Workload Registration
+
+The SPIFFE ID granted to the workload is derived from the 1) service
+account or 2) a configurable pod label.
+
+Service account derived workload registration maps the service account into a
+SPIFFE ID of the form
+`spiffe://<TRUSTDOMAIN>/ns/<NAMESPACE>/sa/<SERVICEACCOUNT>`. For example, if a
+pod came in with the service account `blog` in the `production` namespace, the
+following registration entry would be created:
+
+```
+Entry ID      : 200d8b19-8334-443d-9494-f65d0ad64eb5
+SPIFFE ID     : spiffe://example.org/ns/production/sa/blog
+Parent ID     : spiffe://example.org/k8s-workload-registrar/example-cluster/node
+TTL           : default
+Selector      : k8s:ns:production
+Selector      : k8s:pod-name:example-workload-98b6b79fd-jnv5m
+```
+
+### Label Based Workload Registration
+
+Label based workload registration maps a pod label value into a SPIFFE ID of
+the form `spiffe://<TRUSTDOMAIN>/<LABELVALUE>`. For example if the registrar
+was configured with the `spire-workload` label and a pod came in with
+`spire-workload=example-workload`, the following registration entry would be
+created:
+
+```
+Entry ID      : 200d8b19-8334-443d-9494-f65d0ad64eb5
+SPIFFE ID     : spiffe://example.org/example-workload
+Parent ID     : spiffe://example.org/k8s-workload-registrar/example-cluster/node
+TTL           : default
+Selector      : k8s:ns:production
+Selector      : k8s:pod-name:example-workload-98b6b79fd-jnv5m
+```
+
+Pods that don't contain the pod label are ignored.
+
+## Deployment
+
+The registrar should be deployed as a container in the SPIRE server pod, since
+it talks to SPIRE server via a Unix domain socket. It will need access to a
+shared volume containing the socket file. The registrar will also need access
+to its server keypair and the CA certificate it uses to verify clients.
+
+The following K8S objects are required to set up the validating admission controller:
+* `Service` pointing to the registrar port within the spire-server container
+* `ValidatingWebhookConfiguration` configuring the registrar as a validating admission controller.
+
+Additionally, unless you disable client authentication (`--insecure-skip-verification`), you will need:
```

Comment:
Looks like this is not a flag but a config file option? The config file option doesn't have the `insecure` prefix, which it should.

We should also communicate the impact of setting this flag. If we can do it succinctly in the above config documentation then great, but I have a hunch that it's a little more nuanced than can fit neatly into the table. Perhaps a `Security Considerations` section in that case

Reply:
PS>> If we create a `Security Considerations` section, we should point/link to it from the config option in the table